keyboards/novelkeys/nk87/nk87.c

```diff
@@ -13,46 +13,192 @@
+    {     56,     57,      58,  NO_LED,  NO_LED,  NO_LED,      59,  NO_LED,  NO_LED,  NO_LED,  NO_LED,      60,      61,      62,      47,      55,      66 },
+}, {
+    {   9, 22 }, {  12, 32 }, {  13, 41 }, {  16, 50 }, {  21, 22 }, {  27, 32 }, {  30, 41 }, {  36, 50 }, {  33, 22 }, {  39, 32 },
+    {  42, 41 }, {  48, 50 }, {  45, 22 }, {  51, 32 }, {  53, 41 }, {  60, 50 }, {  57, 22 }, {  63, 32 }, {  65, 41 }, {  72, 50 },
+    {  69, 22 }, {  75, 32 }, {  77, 41 }, {  84, 50 }, {  81, 22 }, {  87, 32 }, {  89, 41 }, {  96, 50 }, {  92, 22 }, {  99, 32 },
+    { 101, 41 }, { 108, 50 }, { 104, 22 }, { 111, 32 }, { 113, 41 }, { 120, 50 }, { 116, 22 }, { 123, 32 }, { 125, 41 }, { 132, 50 },
+    { 128, 22 }, { 135, 32 }, { 137, 41 }, { 143, 50 }, { 140, 22 }, { 147, 32 }, { 149, 41 }, { 192, 60 }, { 152, 22 }, { 159, 32 },
+    { 166, 50 }, { 204, 50 }, { 171, 22 }, { 174, 32 }, { 169, 41 }, { 204, 60 }, {  12, 60 }, {  27, 60 }, {  42, 60 }, {  93, 60 },
+    { 144, 60 }, { 159, 60 }, { 174, 60 }, {   9, 10 }, {  24, 10 }, {  36, 10 }, { 216, 60 }, {  48, 10 }, {  60, 10 }, {  75, 10 },
+    {  87, 10 }, {  99, 10 }, { 111, 10 }, { 126, 10 }, { 138, 10 }, { 150, 10 }, { 162, 10 }, { 192, 32 }, { 192, 22 }, { 177, 10 },
+    { 204, 32 }, { 204, 22 }, { 192, 10 }, { 216, 32 }, { 216, 22 }, { 204, 10 }, { 216, 10 }, { 223,  5 }, { 223, 12 }
+}, {
+    4, 4, 4, 1, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4, 1, 1, 1, 4,
+    1, 1, 1, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 8, 8
+} };
+
 
 /* Indicator LEDS are part of the LED driver
  * Top LED is blue only. LED driver 2 RGB 63 Blue channel
  * Middle LED is blue and red. LED driver 2 RGB 63 Red and Green channel
  * Bottom LED is red only LED driver 2 RGB 48 Blue channel.
  */
-uint8_t CAPS = 0;
-uint8_t FN1 = 0;
-uint8_t FN2 = 0;
-
-bool led_update_kb(led_t led_state) {
-    bool res = led_update_user(led_state);
-    if(res) {
-        if (led_state.caps_lock) {
-            CAPS = 255;
-        } else {
-            CAPS = 0;
-        }
-    }
-    IS31FL3733_set_color( 63+64-1, FN1, FN1, CAPS );
-    IS31FL3733_set_color( 48+64-1, 0, 0, FN2 );
-    return res;
+void indicators_update(uint8_t caps, uint8_t fn1, uint8_t fn2) {
+    rgb_matrix_set_color(87, fn1, fn1, caps);
+    rgb_matrix_set_color(88, 0, 0, fn2);
 }
 
-__attribute__((weak)) layer_state_t layer_state_set_user(layer_state_t state) {
-    if (state & (1UL << 1)) {
-        FN1 = 255;
-    } else {
-        FN1 = 0;
+
+uint8_t caps = 0;
+uint8_t fn1 = 0;
+uint8_t fn2 = 0;
+
+
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
```

Comment:
Will this overwrite `rgb_matrix_indicators_user` if users picked this function over `rgb_matrix_indicators_advanced_user`?

Reply:
No, the `rgb_matrix_indicators_user` will not overwrite the `rgb_matrix_indicators_advanced_user`. Both functions will be called, the only difference is that `rgb_matrix_indicators_user` has no parameters (https://github.com/qmk/qmk_firmware/blob/master/quantum/rgb_matrix/rgb_matrix.c#L432).

I think there is no need to use the `_advanced_` functions in this case since these additional parameters are not used.

Reply:
Switched to use the `rgb_matrix_indicators_kb()` with  the 54ba42e96b719685fbb0f7072470694b2eeea9f6 commit.